### PR TITLE
update endpoint coverage

### PR DESCRIPTION
**Pull request by jc-cb**
adds support for wrapped assets, loans, travel rules, reports, users, and more.

### CODE REVIEW

**Review by nmorgan-cb — APPROVED** [review]

File: rest/exchange_create_loan.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+url = 'https://api.exchange.coinbase.com/loans/open'
+
+timestamp = str(int(time.time()))
+method = 'POST'
+
+url_path = urlparse(url).path
+
+payload = {
+   'loan_id': 'loan_id',
```
Comment (cfluke-cb) [inline rest/exchange_create_loan.py:31]: Do we want working amounts here? Have you tested this endpoint? it lets you set the loan id externally? should that be generated as a uuid here?

File: rest/exchange_create_report.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import json, hmac, hashlib, time, requests, base64, os, sys
+from urllib.parse import urlparse
+
+API_KEY = str(os.environ.get('API_KEY'))
+PASSPHRASE = str(os.environ.get('PASSPHRASE'))
+SECRET_KEY = str(os.environ.get('SECRET_KEY'))
+
+url = 'https://api.exchange.coinbase.com/reports'
+
+report_type = "account"
```
Comment (cfluke-cb) [inline rest/exchange_create_report.py:24]: I would add a comment about what types are allowed

File: rest/exchange_create_travel_rule_entry.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+timestamp = str(int(time.time()))
+idem = str(uuid.uuid4())
+method = 'POST'
+
+url_path = urlparse(url).path
+
+payload = {
+   'address': 'address',
+   'originator_name': 'originator_name',
+   'originator_country': 'originator_country',
```
Comment (cfluke-cb) [inline rest/exchange_create_travel_rule_entry.py:34]: Comment on the format of this. two letter country code?

File: rest/exchange_create_withdrawal_to_payment_method.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+url = 'https://api.exchange.coinbase.com/withdrawals/payment-method'
+
+timestamp = str(int(time.time()))
+method = 'POST'
+
+if len(sys.argv) != 4:
+    exit("Usage: python exchange_create_withdrawal.py <amount> <currency> <payment_method_id>")
+
+amount, currency, payment_method_id = sys.argv[1], sys.argv[2], sys.argv[3]
```
Comment (cfluke-cb) [inline rest/exchange_create_withdrawal_to_payment_method.py:31]: switched up style here from 1 each line to this. i would stay consistent

File: rest/exchange_update_settlement_preference.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+url = f'https://api.exchange.coinbase.com/users/{user_id}/settlement-preferences'
+
+timestamp = str(int(time.time()))
+method = 'POST'
+
+url_path = urlparse(url).path
+
+payload = {
+   'settlement_preference': 'USD',
+   'user_id': 'user_id'
```
Comment (cfluke-cb) [inline rest/exchange_update_settlement_preference.py:36]: i think you want 'user_id': user_id here

File: rest/exchange_transfer_funds_between_profiles.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+url = 'https://api.exchange.coinbase.com/profiles/transfer'
+
+timestamp = str(int(time.time()))
+method = 'POST'
+
+url_path = urlparse(url).path
+
+payload = {
+   'from': 'from',
```
Comment (cfluke-cb) [inline rest/exchange_transfer_funds_between_profiles.py:30]: args

**Review by cfluke-cb — CHANGES_REQUESTED** [review]
small changes

**Review by cfluke-cb — APPROVED** [review]

**Pull request merged**